### ISSUE TITLE
Text generation function only generates random tokens

### ISSUE THREAD

**Issue by alexbui91** [mahmoodlab/CONCH#18]
Hello, thank you for the good work. I am interested in the captioning function. When I tried this function with the following code, the generated text seems like random. It seems like the weights corresponding to the text_encoder is not included inside "pytorch_model.bin".
```
import torch
from huggingface_hub import hf_hub_download
from conch.open_clip_custom import create_model_from_pretrained, get_tokenizer
from PIL import Image

model, preprocess = create_model_from_pretrained("conch_ViT-B-16", checkpoint_path="hf_hub:MahmoodLab/CONCH")
image_path = './docs/roi1.jpg'
image = Image.open(image_path)
image = preprocess(image).unsqueeze(0)
with torch.inference_mode():
    image_embs = model.encode_image(image, proj_contrast=False, normalize=False)
    gen_tex = model.generate(image, seq_len=50, temperature=0.1, generation_type="top_k")

tokenizer = get_tokenizer()
print(tokenizer.decode(gen_tex.tolist()[0]))
```

Output text: "'<start_of_text> dcf juxt oursrinkingwi synchronously irreversible adenohyp 75 theoretical degenerating herds merit discriminationuspid toxicologicvealashi serinebackground injecting precludes replace myring439 jack impliedsectional societies iaslc pca heparan azu acanthoticpassabscess torsionrofen preclud doubtful tai glutamatetazhis naz fusions albicans micromet<end_of_text>'"

**Comment by fedshyvana** [comment]
Hi, yes your understanding is correct. We provided an explanation in the README:

"Note: while the original CONCH model arechitecture also includes a multimodal decoder trained with the captioning loss of CoCa, as additional precaution to ensure that no proprietary data or Protected Health Information (PHI) is leaked untentionally, we have removed the weights for the decoder from the publicly released CONCH weights. The weights for the text encoder and the vision encoder are intact and therefore the results on all key tasks presented in the paper such as image classification and image-text retrieval are not affected. The ability of CONCH to serve as a general purpose encoder for both histopathology images and pathology-related text also remains unaffected."
